contracts/BananaAllocator.sol

```diff
@@ -0,0 +1,235 @@
+    /// @param allocationAmount amount for allocation to recieve
+    /// @param newAllocationAdmins array of addresses to be admins in this specific allocation
+    function addAllocation(
+        string memory allocationName, 
+        uint256 allocationAmount, 
+        address[] memory newAllocationAdmins
+    ) external onlyOwner {
+        totalAllocations += allocationAmount;
+
+        allocationInfo.push(AllocationInfo({
+            allocationName: allocationName,
+            allocationAmount: allocationAmount, 
+            tokensAvailable: 0
+        }));
+
+        uint256 _aid = allocationInfo.length - 1;
+
+        for(uint256 i = 0; i < newAllocationAdmins.length; i++) {
+            allocationInfo[_aid].allocationAdmins[newAllocationAdmins[i]] = true;
+        }
+    
+        emit AllocationAdded(allocationName, allocationAmount, newAllocationAdmins);
+    }
+
+    /// @notice An external function to adjust the amount an allocation has
+    /// @param aid number referencing the applicable allocation
+    /// @param allocationAmount the updated amount to set an allocation to
+    function setAllocation(
+        uint256 aid, 
+        uint256 allocationAmount
+    ) external onlyOwner {
+        uint256 prevAllocationAmount = allocationInfo[aid].allocationAmount;
+
+        allocationInfo[aid].allocationAmount = allocationAmount;
+        if (prevAllocationAmount != allocationAmount) {
+            totalAllocations = totalAllocations.sub(prevAllocationAmount).add(allocationAmount);
+        }
+
+        emit AllocationSet(aid, allocationAmount);
+    }
+
+    /// @notice An external function to add admins to an allocation
+    /// @param aid number referencing the applicable allocation
+    /// @param newAllocationAdmins array of addresses to add to a specific allocation
+    function addAdminsToAllocation(
+        uint256 aid, 
+        address[] memory newAllocationAdmins
+    ) external onlyOwner {
+        for(uint256 i = 0; i < newAllocationAdmins.length; i++) {
+            allocationInfo[aid].allocationAdmins[newAllocationAdmins[i]] = true;
+        }
+
+        emit AdminsAdded(aid, newAllocationAdmins);
+    }
+
+    /// @notice An external function to remove admins from an allocation
+    /// @param aid number referencing the applicable allocation
+    /// @param allocationAdminsToRemove array of addresses to remove from a specific allocation
+    function removeAdminsFromAllocation(
+        uint256 aid, 
+        address[] memory allocationAdminsToRemove
+    ) external onlyOwner {
+        for(uint256 i = 0; i < allocationAdminsToRemove.length; i++) {
+            allocationInfo[aid].allocationAdmins[allocationAdminsToRemove[i]] = false;
+        }
+
+        emit AdminsRemoved(aid, allocationAdminsToRemove);
+    }
+
+    /// @notice An enternal function to transfer dev role of the MasterApe contract
+    function transferDevToOwner() external onlyOwner {
```

Comment:
After further thought, the owner of this contract would be a Timelock, which means we would have to transfer it to the Timelock, then transfer it again to another address. I think we should go with similar ownership transfer logic as the `MasterApeAdmin`

```
    /// @notice Set an address as the pending admin of the MasterApe. The address must accept afterward to take ownership.
    /// @param _pendingMasterApeOwner Address to set as the pending owner of the MasterApe.
    function setPendingMasterApeOwner(address _pendingMasterApeOwner) external onlyOwner {
        pendingMasterApeOwner = _pendingMasterApeOwner;
        emit SetPendingMasterApeOwner(pendingMasterApeOwner);
    }

    /// @notice The pendingMasterApeOwner takes ownership through this call
    /// @dev Transferring MasterApe ownership away from this contract renders this contract useless. 
    function acceptMasterApeOwnership() external {
        require(msg.sender == pendingMasterApeOwner, "not pending owner");
        masterApe.transferOwnership(pendingMasterApeOwner);
        pendingMasterApeOwner = address(0);
    }
```